File: src/cpp/regex.cpp
#include <iostream>
#include <string>
#include <regex>

int main()
{
    std::cout << "Is '^\\d+$' mathed by '123456'?" << std::endl;
    if (std::regex_match("123456", std::regex("^[0-9]+$")))
    {
        std::cout << "yes" << std::endl;
    }
    else
    {
        std::cout << "no" << std::endl;
    }

    std::cout << "Groups of '^([a-z]+)\\-(\\d+)$' found in 'abcdef-12345'" << std::endl;
    std::string test("abcdef-12345");
    std::regex codeRegexp("^([a-z]+)\\-(\\d+)$");
    std::smatch matches;
    if (std::regex_search(test, matches, codeRegexp))
    {
        std::cout << "Found " << matches.size() << " groups." << std::endl;

        for (int groupIndex = 0; groupIndex < matches.size(); groupIndex++)
        {
            std::cout << "[" << groupIndex << "] = " << matches.str(groupIndex) << std::endl;
        }
    }
    return 0;
}

File: src/js/regex.js
console.log("Is '^[0-9]+$' mathed by '123456'?");
if (/^[0-9]+$/.test("123456")) {
    console.log("yes");
}
else {
    console.log("no");
}

console.log("Groups of '^([a-z]+)\\-(\\d+)$' found in 'abcdef-12345'");
var codeRegexp = /^([a-z]+)\-(\d+)$/;
var matches = codeRegexp.exec("abcdef-12345");
console.log(`Found ${matches.length} groups.`);

for (var groupIndex = 0; groupIndex < matches.length; groupIndex++){
    console.log(`[${groupIndex}] = ${matches[groupIndex]}`);
}